commit d25b8fc60fb8c649559924870fe3aaf75e7421d5
Author: Dave Eddy <dave@daveeddy.com>
Date:   2019-05-15T14:14:43-04:00 (5 months ago)
    
    TOOLS-2256 eng check target for rust code
    Reviewed by: Tim Foster <tim.foster@joyent.com>
    Approved by: Tim Foster <tim.foster@joyent.com>

diff --git a/tools/mk/Makefile.targ b/tools/mk/Makefile.targ
index d46b3af..a9e1044 100644
--- a/tools/mk/Makefile.targ
+++ b/tools/mk/Makefile.targ
@@ -45,6 +45,7 @@
 #	check		Checks JavaScript files for lint and style
 #			Checks bash scripts for syntax
 #			Checks SMF manifests for validity against the SMF DTD
+#			Checks rust files using `cargo check`
 #
 #	clean		Removes built files
 #
@@ -101,6 +102,8 @@
 #
 #	JSSTYLE_FILES	JavaScript files to be style-checked
 #
+#	RUST_CODE	Set this to enable rust code syntax checks.
+#
 # You can also override these variables:
 #
 #	BASH		Path to bash (default: "bash")
@@ -108,6 +111,8 @@
 #	BUILD		top-level directory for node binaries, generated docs,
 #			and any other build output (default: "build")
 #
+#	CARGO		Path to cargo (default: "cargo")
+#
 #	CSCOPE_DIRS	Directories to search for source files for the cscope
 #			index. (default: ".")
 #
@@ -141,6 +146,7 @@
 #
 BASH		?= bash
 BASHSTYLE   	?= $(NODE) deps/eng/tools/bashstyle
+CARGO		?= cargo
 CP		?= cp
 CSCOPE		?= cscope
 CSCOPE_DIRS	?= .
@@ -257,8 +263,13 @@ JSSTYLE_TARGET = $(if $(JSSTYLE_FILES), check-jsstyle)
 check-jsstyle:  $(JSSTYLE_EXEC)
 	$(JSSTYLE) $(JSSTYLE_FLAGS) $(JSSTYLE_FILES)
 
+RUST_CHECK_TARGET = $(if $(RUST_CODE), check-rust)
+.PHONY: check-rust
+check-rust:
+	$(CARGO) check
+
 .PHONY: check
-check:: $(ESLINT_TARGET) check-jsl check-json $(JSSTYLE_TARGET) check-bash
+check:: $(ESLINT_TARGET) check-jsl check-json $(JSSTYLE_TARGET) check-bash $(RUST_CHECK_TARGET)
 	@echo check ok
 
 .PHONY: clean
